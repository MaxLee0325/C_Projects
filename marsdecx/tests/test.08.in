2
5 0xfb 0xff 0xff 0xff 0xf0
5 0xfc 0x00 0x00 0x00 0x10
