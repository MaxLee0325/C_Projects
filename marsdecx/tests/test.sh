#!/bin/sh
export LC_ALL=C

echo ======================================================
echo ====================== TEST $1 =======================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ______________Your_output______________ ____________Expected_Output_____________
   #echo +++++++++++++++++++++++++++++++++++++++ ++++++++++++++++++++++++++++++++++++++++
    diff -w -y -W 80 tests/test.$1.out tests/test.$1.expected
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
