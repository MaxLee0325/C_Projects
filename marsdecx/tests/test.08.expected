Line 1, integer 1: 1073741823
Line 2, integer 1: -1073741823
